strings/base_activation.h

```diff
@@ -30,7 +30,10 @@ namespace winrt::impl
     {
         if (winrt_activation_handler)
         {
-            return winrt_activation_handler(*(void**)(&name), guid, result);
+            hresult hr = winrt_activation_handler(*(void**)(&name), guid, result);
+            if (hr != impl::error_not_implemented) {
```

Comment:
I would prefer a different signal. There are cases where it's possible for the real `RoGetActivationFactory()` to return `E_NOTIMPL`. For example, if it successfully loads the server DLL, gets the activation factory, calls `DllGetActivationFactory`, and the server's `DllGetActivationFactory` returns `E_NOTIMPL`, then that error bubbles all the way out from `RoGetActivationFactory`. We don't want this to trigger the fallback code path. We want this to be the definitive answer: It wasn't implemented.

I would suggest that we use the signal `(hr == 0 && !result)`. This would lead to a crash, so we can be pretty confident that nobody is doing it by mistake.

Reply:
I believe you meant `(hr == 0 && !(*result))`.

Reply:
I would go further and suggest a totally new hook, like 'bool try_winrt_activation(void* classId, winrt::guid const& iid, void** factory, int32_t hr) noexcept' which takes precedence over winrt_activation_handler.  then we're not muddying the existing contract.

Reply:
I changed the implementation towards @Scottj1s 's recommendation.